infos.go

```diff
@@ -0,0 +1,30 @@
+// bagdb: Simple datastorage
+// Copyright 2023 billy authors
+// SPDX-License-Identifier: BSD-3-Clause
+
+package billy
+
+// Infos contains a set of statistics about the underlying datastore.
+type Infos struct {
+	Shelves []*ShelfInfos
+}
+
+// ShelfInfos contains some statistics about the data stored in a single shelf.
+type ShelfInfos struct {
+	SlotSize    uint32
+	FilledSlots uint64
+	GappedSlots uint64
+}
+
+// Infos gathers and returns some stats about the database.
+func (db *database) Infos() *Infos {
+	infos := new(Infos)
+	for _, shelf := range db.shelves {
+		infos.Shelves = append(infos.Shelves, &ShelfInfos{
+			SlotSize:    shelf.slotSize,
+			FilledSlots: shelf.count - uint64(len(shelf.gaps)),
+			GappedSlots: uint64(len(shelf.gaps)),
```

Comment:
Think you need the per-shelf `gapsMu` here..? 
```
	gapsMu sync.Mutex // Mutex for operating on 'gaps' and 'count'.
```
Maybe add a method `shelf.Stats()` which delivers e.g. `slotSize, count, gaps` or something like that?

Reply:
ptal